sdk/core/System.ClientModel/src/Internal/TrackingList.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace System.ClientModel.Internal;
+
+internal class TrackingList<T> : List<T>, IList<T>
```

Comment:
What happens if changes are made to a reference type T?  Will this list track those changes?